Towny/src/main/java/com/palmergames/bukkit/towny/utils/BorderUtil.java

```diff
@@ -175,4 +187,94 @@ public static boolean allowedMove(Block block, Block blockTo, @Nullable Player p
 			&& !currentTownBlock.hasResident() && !destinationTownBlock.hasResident();
 		
 	}
+
+	private static final int[][] DIRECTIONS = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
+
+	@ApiStatus.Internal
+	public static @NotNull FloodfillResult getFloodFillableCoords(final @NotNull Town town, final @NotNull WorldCoord origin) {
```

Comment:
The only other thing I can see being useful here is being able to test if the town has enough claims to even claim the filled area. We would have to take into account towns with unlimited claims as well.